DOCS-8523 Add Multisource Querying Documentation (#24312)
DataDog/documentation: content/en/cloud_cost_management/multisource_querying/_index.md

---
title: Multisource Querying
is_beta: true
description: Learn how to use Multisource Querying to query costs across multiple providers in Cloud Cost Management.
further_reading:
- link: "/cloud_cost_management/"
  tag: "Documentation"
  text: "Learn about Cloud Cost Management"
- link: "/cloud_cost_management/container_cost_allocation"
  tag: "Documentation"
  text: "Learn about Container Cost Allocation"
---

{{< site-region region="gov" >}}
<div class="alert alert-warning">Cloud Cost Management is not supported for your selected <a href="/getting_started/site">Datadog site</a> ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

<div class="alert alert-warning">Multisource Querying is in private beta. To request access, contact your Customer Success Manager.</div>

## Overview

> After you start ingesting your [AWS][1], [Azure][2], [Google Cloud][3], [SaaS][4], or [Datadog costs][5] in [Cloud Cost Management][6], you can flexibly query costs across providers. Multisource Querying enables you to query costs across multiple providers using consistent and standardized tags instead of needing to build multiple queries for each provider.

Reviewer: ```suggestion
After you start ingesting your [AWS][1], [Azure][2], [Google Cloud][3], [SaaS][4], or [Datadog costs][5] in [Cloud Cost Management][6], you can flexibly query costs across providers. Multisource Querying enables you to query costs across multiple providers using consistent and standardized tags instead of building multiple queries for each provider.
```

Use Multisource Querying to build cost views, understand the total cost of service ownership, and alert on cost changes and trends across the [**Analytics** page][6], [dashboards][7], [notebooks][8], and [cost monitors][9].

## Setup

To use Multisource Querying, ensure you meet the following requirements:

1. You have configured [Cloud Cost Management][10] and are actively ingesting costs in Datadog.
1. Your cost data is in a single currency. Multiple currencies are not supported. 

## Query your cost data

> You can select multiple providers in the `Provider` field on the [**Analytics** page][6].

Reviewer: ```suggestion
You can select multiple providers in the **Provider** field on the [**Analytics** page][6].
```

{{< img src="cloud_cost/multisource_querying/provider.png" alt="The Provider field below the search query on the Cloud Cost Analytics page" style="width:40%;" >}}

> Dropdown filters like `Provider` maintain flexibility and streamline the process of creating a search query so you can refine your cost data. To add a filter, click **+ Filter**. 

Reviewer: ```suggestion
Dropdown filters like **Provider** maintain flexibility and streamline the process of creating a search query so you can refine your cost data. To add a filter, click **+ Filter**. 
```

{{< img src="cloud_cost/multisource_querying/filters.png" alt="The Service and Team filters highlighted below the search query on the Cloud Cost Analytics page" style="width:100%;" >}}

Click **Refine Results** to access the following options and filter your cost data.

Usage Charges Only
: Examine costs impacted by engineering teams, excluding credits, fees, and taxes.

Complete Days Only

> : Exclude the past two days of cost data that are incomplete.

Reviewer: ```suggestion
: Exclude the past two days of cost data, which are incomplete.
```

Total Cost
: Filter the data to view costs within a specific cost range.

Dollar Change
: Only display cost changes within a specified dollar change range.

Percent Change
: Only display cost changes within a specified percentage change range.

{{< img src="cloud_cost/multisource_querying/refine_results.png" alt="Additional options to refine your cost data on the Cloud Cost Analytics page" style="width:100%;" >}}

## Visualize your cost data

With Multisource Querying, you can create visualizations using cost data across providers in your [dashboards][11].

{{< img src="cloud_cost/multisource_querying/cost_overview.png" alt="A dashboard in Datadog displaying Cloud Cost Management data from multiple providers such as Snowflake, Azure, Google Cloud, AWS, and more" style="width:100%;" >}}

> ## Data Collected

Reviewer: ```suggestion
## Data collected
```
I feel like we're inconsistent about this (and Further reading / Further Reading) so I don't know whether you want to fix it or not, just thought I'd flag it

| `ecs_cluster_name` | The name of the ECS cluster hosting a workload. |
| `kube_cluster_name` | The name of the Kubernetes cluster hosting a workload. |
| `orchestrator` | The container orchestrator (such as `kubernetes` or `ecs`). |

### Tag enrichment

Cloud Cost Management enriches all provider cost data with tags that adhere to the [FinOps FOCUS specification][12]. FOCUS™ is a technical specification that normalizes cost and usage billing data across cloud vendors.

FOCUS tags allow you to query similar concepts across providers. For example, if you'd like to see cost per account across AWS and Azure, you are not required to create two queries (one for AWS costs grouped by `aws_member_account_name` and one for Azure costs grouped by `subscriptionname`). You can use one search query that filters to AWS and Azure costs grouped by `subaccountname`.

Cloud Cost Management adds lowercase versions of the specification Column IDs to all cost metrics.

The following FOCUS tags are available in Cloud Cost Management:

| Tag Name | Tag Description |
|---|---|
| `providername` | The name of the entity that made the resources or services available for purchase. |

> | `servicename` | An offering that can be purchased from a provider (e.g., cloud virtual machine, SaaS database, professional services from a systems integrator). |

Reviewer: ```suggestion
| `servicename` | An offering that can be purchased from a provider (for example, cloud virtual machine, SaaS database, professional services from a systems integrator). |
```

| `billingaccountid` | The identifier assigned to a billing account by the provider. |
| `billingaccountname` | The display name assigned to a billing account. |

> | `subaccountid` | An ID assigned to a grouping of resources or services, often used to manage access and/or cost. |

Reviewer: ```suggestion
| `subaccountid` | An ID assigned to a grouping of resources or services, often used to manage access or cost. |
```

> | `subaccountname` | A name assigned to a grouping of resources or services, often used to manage access and/or cost. |

Reviewer: ```suggestion
| `subaccountname` | A name assigned to a grouping of resources or services, often used to manage access or cost. |
```
